commit/factory.go

```diff
@@ -90,7 +90,7 @@ func NewPluginFactory(
 	}
 }
 
-func (p *PluginFactory) NewReportingPlugin(config ocr3types.ReportingPluginConfig,
+func (p *PluginFactory) NewReportingPlugin(ctx context.Context, config ocr3types.ReportingPluginConfig,
```

Comment:
What is this context? Is there a proper lifecycle context now?

This is called by `chainlink` code, so the full integration test will break as a result.